CONTRIBUTING.md

```diff
@@ -636,3 +637,33 @@ $$\alpha = \frac{\beta}{\gamma}$$
 
 Example of auto-built documentation for `webviz-config` can be seen
 [here on github](https://equinor.github.io/webviz-config/).
+
+
+## Deprecate plugins or arguments
+
+Plugins can be marked as deprecated by using the `@deprecated_plugin(short_message, long_message)` decorator.
+
+```python
+@deprecated_plugin("This message is shown to the end user in the browser.", "This message is shown in the documentation of the plugin.")
+class MyPlugin(WebvizPluginABC):
+    ...
+```
+
+Plugin arguments can be marked as deprecated by using the `@deprecated_plugin_arguments(arguments={})` decorator in front of the `__init__` function.
+Arguments can either be marked as deprecated in any case or their values can be checked within a function which then returns a tuple containing a 
+boolean stating if they are deprecated with the given value and a short string shown to the end user in the browser and a long string shown in the 
+plugin's documentation.
+
+```python
+class MyPlugin(WebvizPluginABC):
+    ...
+    @deprecated_plugin_arguments(arguments={"arg1": ("This message is shown to the end user in the browser.", "This message is shown in the documentation of the plugin."), "arg2": check_deprecation})
+    def __init__(self, arg1: str, arg2: int, arg3: int):
+        ...
+
+def check_deprecation(arg_name: str, arg_value: Any, other_args: Dict[str, Any]) -> Tuple[bool, str, str]:
```

Comment:
```suggestion
def check_deprecation(arg2: int, arg3: int):
```
Is this easy/preferred, i.e. that the decorator injects the necessary values to the `check_deprecation` function? Benefits are shorter signature + explicit types (avoid `Any`).

Reply:
Hmm, I see your point. My thought was that there might be cases where you also want to access other, non-deprecated variables in order to check if there is a deprecation. Or is your suggestion to pass on all arguments from the __init__ method?